packages/experimental/Divider/src/Divider.styling.ts

```diff
@@ -0,0 +1,149 @@
+        minWidth: tokens.minWidth,
+        maxWidth: tokens.maxWidth,
+        minHeight: tokens.minHeight,
+        maxHeight: tokens.maxHeight,
+        padding: tokens.padding,
+        paddingStart: tokens.paddingStart,
+        paddingEnd: tokens.paddingEnd,
+        paddingHorizontal: tokens.paddingHorizontal,
+        paddingVertical: tokens.paddingVertical,
+        ...(tokens.vertical
+          ? {
+              paddingVertical: tokens.insetSize,
+              height: '100%',
+            }
+          : {
+              paddingHorizontal: tokens.insetSize,
+            }),
+      },
+    }),
+    [
+      tokens.vertical,
+      tokens.insetSize,
+      tokens.minHeight,
+      tokens.maxHeight,
+      tokens.minWidth,
+      tokens.maxWidth,
+      tokens.padding,
+      tokens.paddingHorizontal,
+      tokens.paddingVertical,
+      tokens.paddingStart,
+      tokens.paddingEnd,
+    ],
+  );
+
+  const beforeLineProps: ViewProps = useMemo(
+    () => ({
+      style: {
+        flexBasis: tokens.minLineSize,
+        flex: tokens.flexBefore,
+        borderColor: tokens.color || tokens.lineColor,
+        borderStyle: 'solid',
+        ...(tokens.vertical
+          ? { borderLeftWidth: tokens.thickness, minHeight: tokens.minLineSize }
+          : { borderTopWidth: tokens.thickness, minWidth: tokens.minLineSize }),
+      },
+    }),
+    [tokens.color, tokens.flexBefore, tokens.lineColor, tokens.minLineSize, tokens.thickness, tokens.vertical],
+  );
+
+  const afterLineProps: ViewProps = useMemo(
+    () => ({
+      style: {
+        flexBasis: tokens.minLineSize,
+        flex: tokens.flexAfter,
+        borderColor: tokens.color || tokens.lineColor,
+        borderStyle: 'solid',
+        ...(tokens.vertical
+          ? { borderLeftWidth: tokens.thickness, minHeight: tokens.minLineSize }
+          : { borderTopWidth: tokens.thickness, minWidth: tokens.minLineSize }),
+      },
+    }),
+    [tokens.color, tokens.flexAfter, tokens.lineColor, tokens.minLineSize, tokens.thickness, tokens.vertical],
+  );
+
+  const wrapperProps: ViewProps = useMemo(
+    () => ({
+      style: {
+        flex: 0,
+        ...(tokens.vertical ? { paddingVertical: tokens.contentPadding } : { paddingHorizontal: tokens.contentPadding }),
+      },
+    }),
+    [tokens.contentPadding, tokens.vertical],
+  );
+
+  const textProps: TextProps = useMemo(
+    () => ({
+      style: {
+        color: tokens.color || tokens.contentColor,
+      },
+    }),
+    [tokens.color, tokens.contentColor],
+  );
+
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const iconProps: IconProps = useMemo(() => (icon ? getIconProps(tokens, icon) : {}), [icon, tokens.color, tokens.contentColor]);
+
+  return { rootProps, beforeLineProps, afterLineProps, wrapperProps, textProps, iconProps };
+};
+
+export const colorsFromAppearance = (appearance: DividerAppearance, theme: Theme): Pick<DividerTokens, 'contentColor' | 'lineColor'> => {
```

Comment:
Instead of `colorsFromApperance`, could we instead get the appearance from the theme and call this something like `DividerColorsFromTheme` ? Also add a small amount of documentation why this needs to be a separate function?